diff --git a/src/libscotch/context.c b/src/libscotch/context.c
index fbd8cbc..c081a26 100644
--- a/src/libscotch/context.c
+++ b/src/libscotch/context.c
@@ -88,7 +88,7 @@ int
 contextOptionsInit (
 Context * const             contptr)
 {
-  return (contextValuesInit (contptr, &contextvaluesdat, sizeof (contextvaluesdat),
-                             CONTEXTOPTIONNUMNBR, (byte *) &contextvaluesdat.vinttab - (byte *) &contextvaluesdat,
-                             CONTEXTOPTIONDBLNBR, (byte *) &contextvaluesdat.vdbltab - (byte *) &contextvaluesdat));
+  size_t offset = sizeof(contextvaluesdat.vinttab);
+  return contextValuesInit(contptr, &contextvaluesdat, sizeof(contextvaluesdat), CONTEXTOPTIONNUMNBR,
+                           offset, CONTEXTOPTIONDBLNBR, offset + sizeof(contextvaluesdat.vinttab));
 }
