diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1c05a82..44a9db6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -95,9 +95,11 @@ if(NOT INTSIZE STREQUAL "")
   if(INTSIZE STREQUAL "32")
     message(STATUS "Integer size is 32 bits")
     add_definitions(-DINTSIZE32)
+    add_definitions(-DIDXSIZE32)
   elseif(INTSIZE STREQUAL "64")
     message(STATUS "Integer size is 64 bits")
     add_definitions(-DINTSIZE64)
+    add_definitions(-DIDXSIZE64)
   else()
     message(FATAL_ERROR "Invalid integer size value")
   endif()
