diff --git a/src/libscotch/dummysizes.c b/src/libscotch/dummysizes.c
index fbd7b52..c5238e7 100644
--- a/src/libscotch/dummysizes.c
+++ b/src/libscotch/dummysizes.c
@@ -92,7 +92,9 @@
 #endif /* SCOTCH_NAME_SUFFIX */
 #include "module.h"
 #include "common.h"
+#define DUMMYSIZES_C
 #include "parser.h"
+#undef DUMMYSIZES_C
 #include "graph.h"
 #include "geom.h"
 #include "mesh.h"

diff --git a/src/libscotch/parser.c b/src/libscotch/parser.c
index a1dc4d8..a537256 100644
--- a/src/libscotch/parser.c
+++ b/src/libscotch/parser.c
@@ -79,7 +79,7 @@
 */
 
 static StratTab             stratdummytab = { NULL, NULL, NULL }; /* Dummy strategy table for the dummy empty object       */
-Strat                       stratdummy = { &stratdummytab, STRATNODEEMPTY }; /* Dummy empty object for offset computations */
+SCOTCH_EXPORT Strat                       stratdummy = { &stratdummytab, STRATNODEEMPTY }; /* Dummy empty object for offset computations */
 
 /**************************/
 /*                        */
diff --git a/src/libscotch/parser.h b/src/libscotch/parser.h
index fa07ef3..e3060a0 100644
--- a/src/libscotch/parser.h
+++ b/src/libscotch/parser.h
@@ -62,6 +62,10 @@
 
 /*+ Strategy node types. +*/
 
+#ifndef DUMMYSIZES_C
+#include <scotch_export.h>
+#endif
+
 typedef enum StratNodeType_ {
   STRATNODECONCAT,                                /*+ Concatenation node       +*/
   STRATNODECOND,                                  /*+ Condition node           +*/
@@ -245,7 +249,9 @@ typedef struct ParserLocation_ {
 **  The external declarations.
 */
 
-extern Strat                stratdummy;           /*+ Dummy empty strategy node +*/
+#ifndef DUMMYSIZES_C
+SCOTCH_EXPORT extern Strat                stratdummy;           /*+ Dummy empty strategy node +*/
+#endif
 
 /*
 **  The function prototypes.
diff --git a/src/libscotch/hgraph_order_st.c b/src/libscotch/hgraph_order_st.c
index e9fabd4..2e620a2 100644
--- a/src/libscotch/hgraph_order_st.c
+++ b/src/libscotch/hgraph_order_st.c
@@ -239,7 +239,7 @@ static StratParamTab        hgraphorderstcondtab[] = { /* Graph condition parame
                               { STRATNODENBR,         STRATPARAMINT,    NULL,
                                 NULL, NULL, NULL } };
 
-StratTab                    hgraphorderststratab = { /* Strategy tables for graph ordering methods */
+SCOTCH_EXPORT StratTab                    hgraphorderststratab = { /* Strategy tables for graph ordering methods */
                               hgraphorderstmethtab,
                               hgraphorderstparatab,
                               hgraphorderstcondtab };
diff --git a/src/libscotch/hgraph_order_st.h b/src/libscotch/hgraph_order_st.h
index 66392d5..6ed6644 100644
--- a/src/libscotch/hgraph_order_st.h
+++ b/src/libscotch/hgraph_order_st.h
@@ -56,6 +56,8 @@
 
 /*+ Method types. +*/
 
+#include <scotch_export.h>
+
 typedef enum HgraphOrderStMethodType_ {
   HGRAPHORDERSTMETHBL = 0,                        /*+ Block splitting post-processing       +*/
   HGRAPHORDERSTMETHCC,                            /*+ Splitting of connected components     +*/
@@ -73,7 +75,7 @@ typedef enum HgraphOrderStMethodType_ {
 **  The external declarations.
 */
 
-extern StratTab             hgraphorderststratab;
+SCOTCH_EXPORT extern StratTab             hgraphorderststratab;
 
 /*
 **  The function prototypes.
diff --git a/src/libscotch/bgraph_bipart_st.c b/src/libscotch/bgraph_bipart_st.c
index b5c2b98..f461581 100644
--- a/src/libscotch/bgraph_bipart_st.c
+++ b/src/libscotch/bgraph_bipart_st.c
@@ -230,7 +230,7 @@ static StratParamTab        bgraphbipartstcondtab[] = { /* Active graph conditio
                               { STRATNODENBR,        STRATPARAMINT,    NULL,
                                 NULL, NULL, NULL } };
 
-StratTab                    bgraphbipartststratab = { /* Strategy tables for graph bipartitioning methods */
+SCOTCH_EXPORT StratTab                    bgraphbipartststratab = { /* Strategy tables for graph bipartitioning methods */
                               bgraphbipartstmethtab,
                               bgraphbipartstparatab,
                               bgraphbipartstcondtab };
diff --git a/src/libscotch/bgraph_bipart_st.h b/src/libscotch/bgraph_bipart_st.h
index 71a7cbf..06332ec 100644
--- a/src/libscotch/bgraph_bipart_st.h
+++ b/src/libscotch/bgraph_bipart_st.h
@@ -57,6 +57,8 @@
 
 /** Method types. **/
 
+#include <scotch_export.h>
+
 typedef enum BgraphBipartStMethodType_ {
   BGRAPHBIPARTSTMETHBD = 0,                       /*+ Band                   +*/
   BGRAPHBIPARTSTMETHDF,                           /*+ Diffusion              +*/
@@ -73,7 +75,7 @@ typedef enum BgraphBipartStMethodType_ {
 **  The external declarations.
 */
 
-extern StratTab             bgraphbipartststratab;
+SCOTCH_EXPORT extern StratTab             bgraphbipartststratab;
 
 /*
 **  The function prototypes.

diff --git a/src/libscotch/kgraph_map_st.c b/src/libscotch/kgraph_map_st.c
index 6abb229..b1914ba 100644
--- a/src/libscotch/kgraph_map_st.c
+++ b/src/libscotch/kgraph_map_st.c
@@ -238,7 +238,7 @@ static StratParamTab        kgraphmapstcondtab[] = { /* Graph condition paramete
                               { STRATNODENBR,        STRATPARAMINT,    NULL,
                                 NULL, NULL, NULL } };
 
-StratTab                    kgraphmapststratab = { /* Strategy tables for graph mapping methods */
+SCOTCH_EXPORT StratTab                    kgraphmapststratab = { /* Strategy tables for graph mapping methods */
                               kgraphmapstmethtab,
                               kgraphmapstparatab,
                               kgraphmapstcondtab };
diff --git a/src/libscotch/kgraph_map_st.h b/src/libscotch/kgraph_map_st.h
index be98f82..bbba461 100644
--- a/src/libscotch/kgraph_map_st.h
+++ b/src/libscotch/kgraph_map_st.h
@@ -60,6 +60,8 @@
 
 /*+ Method types. +*/
 
+#include <scotch_export.h>
+
 typedef enum KgraphMapStMethodType_ {
   KGRAPHMAPSTMETHBD = 0,                          /*+ Band (strategy)               +*/
   KGRAPHMAPSTMETHCP,                              /*+ Old mapping copy              +*/
@@ -75,7 +77,7 @@ typedef enum KgraphMapStMethodType_ {
 **  The external declarations.
 */
 
-extern StratTab             kgraphmapststratab;
+SCOTCH_EXPORT extern StratTab             kgraphmapststratab;
 
 /*
 **  The function prototypes.

diff --git a/src/libscotch/vgraph_separate_st.c b/src/libscotch/vgraph_separate_st.c
index a218fb2..7230f68 100644
--- a/src/libscotch/vgraph_separate_st.c
+++ b/src/libscotch/vgraph_separate_st.c
@@ -213,7 +213,7 @@ static StratParamTab        vgraphseparatestcondtab[] = { /* Graph condition par
                               { STRATNODENBR,        STRATPARAMINT,    NULL,
                                 NULL, NULL, NULL } };
 
-StratTab                    vgraphseparateststratab = { /* Strategy tables for vertex separation methods */
+SCOTCH_EXPORT StratTab                    vgraphseparateststratab = { /* Strategy tables for vertex separation methods */
                               vgraphseparatestmethtab,
                               vgraphseparatestparatab,
                               vgraphseparatestcondtab };
diff --git a/src/libscotch/vgraph_separate_st.h b/src/libscotch/vgraph_separate_st.h
index 8a4d2c6..b95f82b 100644
--- a/src/libscotch/vgraph_separate_st.h
+++ b/src/libscotch/vgraph_separate_st.h
@@ -60,6 +60,8 @@
 
 /*+ Method types. +*/
 
+#include <scotch_export.h>
+
 typedef enum VgraphSeparateStMethodType_ {
   VGRAPHSEPASTMETHBD = 0,                         /*+ Banding strategy         +*/
   VGRAPHSEPASTMETHES,                             /*+ Edge separation strategy +*/
@@ -76,7 +78,7 @@ typedef enum VgraphSeparateStMethodType_ {
 **  The external declarations.
 */
 
-extern StratTab             vgraphseparateststratab;
+SCOTCH_EXPORT extern StratTab             vgraphseparateststratab;
 
 /*
 **  The function prototypes.
